packages/salesforcedx-vscode-core/src/metadataSupport/metadataHoverProvider.ts

```diff
@@ -0,0 +1,308 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import * as vscode from 'vscode';
+import { VALID_METADATA_TYPES } from '../constants';
+import { MetadataDocumentationService } from './metadataDocumentationService';
+
+/**
+ * Provides hover documentation for Salesforce metadata types in XML files
+ */
+export class MetadataHoverProvider implements vscode.HoverProvider {
+  private documentationService: MetadataDocumentationService;
+
+  constructor() {
```

Comment:
I think it's awkward to have both a `new` (that kinda does nothing) and then a required `initialize`.

What if `provideHover` first checked for initialization and then initialized the docService only if it wasn't already?

That way, you defer that big file load until the first time someone actually hovers?

Alternatively, if you really want to do everything as soon as it loads, we could make the ctor private and have the `initialize` method return the class instance so that that's the only way to instantiate it.